src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -21,71 +41,871 @@ const ActorInit En_Rd_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE3A8C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE3B18.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE3C20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE3C98.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK0, 0x00, 0x09, 0x09, 0x10, COLSHAPE_CYLINDER },
+    { 0x01, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x05, 0x01 },
+    { 20, 70, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE3DE4.s")
+static DamageTable sDamageTable = {
+    0x00, 0xF2, 0x00, 0x00, 0x00, 0x00, 0xF2, 0x10, 0xF1, 0xF2, 0xF4, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0xE4, 0x60, 0xD3, 0x00, 0x00, 0xF1, 0xF4, 0xF2, 0xF2, 0xF8, 0xF4, 0x00, 0x00, 0xF4, 0x00,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE3ECC.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(unk_4C, 2000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -3500, ICHAIN_STOP),
+};
 
+static Vec3f D_80AE4918 = { 0.0f, 0.0f, 0.0f };
+
+// I'm guessing these are primitive and environment colors that go unused
+static Color_RGBA8_n D_80AE4924 = { 200, 200, 255, 255 };
+static Color_RGBA8_n D_80AE4928 = { 0, 0, 255, 0 };
+
+static Vec3f D_80AE492C = { 0.0f, 0.0f, 0.0f };
+static Color_RGBA8_n D_80AE4938 = { 200, 200, 255, 255 };
+static Color_RGBA8_n D_80AE493C = { 0, 0, 255, 0 };
+
+static Vec3f D_80AE4940 = { 300.0f, 0.0f, 0.0f };
+static Vec3f D_80AE494C = { 300.0f, 0.0f, 0.0f };
+static Vec3f D_80AE4958 = { 0.25f, 0.25f, 0.25f };
+
+extern SkeletonHeader D_06003DD8;
+extern AnimationHeader D_06004268;
+extern AnimationHeader D_060046F8;
+extern AnimationHeader D_06004ADC;
+extern AnimationHeader D_06004F94;
+extern AnimationHeader D_060057AC;
+extern AnimationHeader D_06005D98;
+extern AnimationHeader D_06006E88;
+extern AnimationHeader D_060074F0;
+extern AnimationHeader D_06008040;
+extern AnimationHeader D_060087D0;
+extern SkeletonHeader D_0600E778;
+extern AnimationHeader D_0600EFDC;
+
+void EnRd_SetupAction(EnRd* this, EnRdActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnRd_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnRd* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->actor.unk_1F = 0;
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    this->unk_310 = this->unk_30E = 0;
+    this->actor.posRot2.pos = this->actor.posRot.pos;
+    this->actor.posRot2.pos.y += 50.0f;
+    this->actor.colChkInfo.mass = 0xFE;
+    this->actor.colChkInfo.health = 8;
+    this->unk_314 = this->unk_31D = 0xFF;
+    this->unk_312 = (THIS->actor.params & 0xFF00) >> 8;
```

Comment:
Using `thisx` in this function would probably avoid the need for any inline `THIS`